pkg/cmd/cli/cmd/importer/dockercompose.go

```diff
@@ -0,0 +1,182 @@
+package importer
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/kubectl"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/client"
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	configcmd "github.com/openshift/origin/pkg/config/cmd"
+	"github.com/openshift/origin/pkg/generate/app"
+	appcmd "github.com/openshift/origin/pkg/generate/app/cmd"
+	"github.com/openshift/origin/pkg/generate/dockercompose"
+)
+
+const (
+	dockerComposeLong = `
+Import a Docker Compose file as OpenShift objects
```

Comment:
More like we can't keep adding function (I may remove compose setup and
move it to init).  I like templates in new-app - but app.json and compose
aren't new-app sort of things.

On Sun, May 1, 2016 at 9:41 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/cmd/cli/cmd/importer/dockercompose.go
> <https://github.com/openshift/origin/pull/7690#discussion_r61697531>:
>
> > +	"k8s.io/kubernetes/pkg/kubectl"
> > +	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
> > +	"k8s.io/kubernetes/pkg/runtime"
> > +
> > +	"github.com/openshift/origin/pkg/client"
> > +	cmdutil "github.com/openshift/origin/pkg/cmd/util"
> > +	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
> > +	configcmd "github.com/openshift/origin/pkg/config/cmd"
> > +	"github.com/openshift/origin/pkg/generate/app"
> > +	appcmd "github.com/openshift/origin/pkg/generate/app/cmd"
> > +	"github.com/openshift/origin/pkg/generate/dockercompose"
> > +)
> > +
> > +const (
> > +	dockerComposeLong = `
> > +Import a Docker Compose file as OpenShift objects
>
> i'd still rather see us just get rid of process rather than new-app
> shedding template function (depending what function you're talking about
> new-app shedding in (1).
>
> At least i think that's what i'd prefer. i do admit new-app not having to
> go through the "does this name match a potential template" dance would be
> nice.
>
> anyway otherwise i think i'm convinced.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7690/files/b85cc800bdcb1e7db8f8c37dd3e85c21b5023f0a#r61697531>
>